merkle/tree_hasher.go

```diff
@@ -40,6 +40,7 @@ var hashTypes = map[string]TreeHasher{
 }
 
 // Factory supports fetching custom hashers based on tree types.
+// TODO(codingllama): Get rid of Factory and hashTypes. It's never used by production code.
```

Comment:
My 2c: if you want an specific hasher, like RFC 6962, you just do `rfc6962.New()` (or whatever equivalent). If you want a config-based hasher, `trees.Hasher` is your friend. In either case, I find the arbitrary-string-based factory approach to not be beneficial, as the strings here do not reflect any kind of external configuration (plus, if you know the string constant you want, you might as well just create the hasher type directly).